libarchive/archive_read_support_format_tar.c

```diff
@@ -1677,7 +1677,9 @@ header_ustar(struct archive_read *a, struct tar *tar,
 
 	/* Copy name into an internal buffer to ensure null-termination. */
 	const char *existing_pathname = archive_entry_pathname(entry);
-	if (existing_pathname == NULL || existing_pathname[0] == '\0') {
```

Comment:
Here's the real problem:  In pax format, we first read the pax extensions and then read the legacy ustar header.  If the pax extensions specified a pathname, we should ignore the pathname in the legacy header.  (Among other problems, the legacy header has no indication of charset.)

I think the right fix here might be to introduce the `pathname_is_set()` function that @dunhor suggested at one point.  As a quick-and-dirty workaround, let's try explicitly checking the wide pathname and see if that actually fixes the test failure.

Reply:
Note:  Probably we have the same problem with link paths in header_common.

Reply:
Yep, we did.  Fixed those also.